datafusion/optimizer/src/optimize_projections/mod.rs

```diff
@@ -234,9 +241,18 @@ fn optimize_projections(
                     // refers to that schema
                     let required_exprs =
                         required_indices.get_required_exprs(&input_schema);
-                    let window_child =
-                        add_projection_on_top_if_helpful(window_child, required_exprs)?
-                            .data;
+
+                    if !config
+                        .options()
+                        .optimizer
+                        .optimize_projections_preserve_existing_projections
+                    {
+                        window_child = add_projection_on_top_if_helpful(
+                            window_child,
+                            required_exprs,
+                        )?
+                        .data;
+                    }
```

Comment:
It will be cleaner to just pass the config into `add_projection_on_top_if_helpful` and have it perform the check, instead of having a 4 line if statement wrapping each time `add_projection_on_top_if_helpful` is called

Reply:
Updated - thank you!